pkg/resource/function/hooks.go

```diff
@@ -416,6 +413,19 @@ func customPreCompare(
 				delta.Add("Spec.Code.ImageURI", a.ko.Spec.Code.ImageURI, b.ko.Spec.Code.ImageURI)
 			}
 		}
+		expected, _ := json.Marshal(a.ko.Spec.CodeS3SHA256)
```

Comment:
I'm not seeing where `Spec.CodeS3SHA256` is ever being set to a value. How about calculating the S3 content's SHA256 manually and comparing it to `Status.CodeSHA256` to determine if the function's S3-backed code has changed?

Reply:
When the user provides the updated deployment package they have to manually calculate the `CodeSHA256` for their code, which they can do by running this command `sha256sum filename.zip |cut -f1 -d\ | xxd -r -p | base64`.  We then use this provided CodeSHA256 and compare it with `Status.CodeSHA256` to see the changes/update.